README.md

```diff
@@ -4,111 +4,214 @@ An API for validating survey schemas.
 
+### Running the Ajv (server) version of validator
 
-## Testing
+Then run the Ajv server:
+``` shell
+make start-ajv
+```
 
-By default, all schemas in the `tests/schemas/valid` directory will be evaluated as part of the unit tests.
-Any errors in these schemas will cause a failure.
+This defaults to running on port `5002` set `AJV_VALIDATOR_PORT` in your .env file if you need to change this.
 
-To run the app's unit tests:
+Running the Ajv server returns either an empty json response when the questionnaire is valid, or a response containing an "errors" key. The errors are ordered by their path length and with first error message being the deepest path into the schema and should represent the best match for the questionnaire which has been posted.
 
+To stop the Ajv server:
+``` shell
+make stop-ajv
 ```
+
+### Testing and running against local schemas
+
+By default, all schemas in the `tests/schemas/valid` and `tests/schemas/invalid` directories will be evaluated as part of the unit tests. Any errors in these schemas will cause a failure.
+
+To run the app's unit tests:
+``` shell
 make test-unit
 ```
 
-To run the app's unit tests and ajv validator tests:
+Make sure you don't already have Ajv running on localhost:5002 by running `lsof -i tcp:5002` if you do make a note of the PID and then run `kill -9 PID`, replacing "PID" with the process id from the previous command.
 
+Run the Ajv validator tests:
+``` shell
+make test-ajv
 ```
+
+To run the app's unit tests and Ajv validator tests:
+``` shell
 make test
 ```
 
-To test the apps functionality:
-```
+#### Test the local validator app against runner schemas
+
+Spin validator up with:
+``` shell
 make run
 ```
 
-Then, in another terminal window/tab, navigate to a checked out copy of eq-survey-runner:
+Then, in another terminal, navigate to a checked out copy of `https://github.com/ONSdigital/eq-questionnaire-runner` and run:
+``` shell
+make validate-test-schemas
 ```
-make test
+This will run the validator against all runner test schemas.
+
+Or you can run it against a specific runner schema, to do this:
+- set the following vars passing them into the following command:
+    - `SCHEMA_PATH` to the path of the schema file (defaults to `./schemas/test/en/`)
+    - `SCHEMA` to the schema file name without the `.json`
+- then run:
+``` shell
+make validate-test-schema SCHEMA=test_checkbox SCHEMA_PATH=./schemas/
 ```
 
-# Installing node dependencies
+## Formatting/linting json
 
-In the eq-schema-validator directory, install node version manager (nvm) and node using the following commands:
+Run the following to format all json files in the schemas directory:
 
-```
-brew install nvm
-nvm install
+``` shell
+make format
 ```
 
-Install node dependencies:
+Run the following to lint all schemas, test schemas and Ajv files in the repo:
 
-```
-npm install
+``` shell
+make lint
 ```
 
-To run the ajv validator tests:
+## Running with Docker
```

Comment:
I missed this with my original review sorry, but I think it makes sense to logically put the Docker installation section before the sections that make use of docker commands etc, so it's like a prerequisite sort of thing

Reply:
I can move it to the top but I think this is the first instance where we are using docker to run validator? Unless you mean the parts that talk about spinning up runner?

Reply:
Yeah, I meant at the top so it was bunched in with the other stuff we install with `brew`, but I've thought a bit more about this, and it's probably fine where it is now, so will resolve 👍